analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/HardcodedIpAddressBase.cs

```diff
@@ -34,59 +34,61 @@ public abstract class HardcodedIpAddressBase<TSyntaxKind, TLiteralExpression> :
         where TSyntaxKind : struct
         where TLiteralExpression : SyntaxNode
     {
-        protected const string DiagnosticId = "S1313";
+        private const string DiagnosticId = "S1313";
         private const string MessageFormat = "Make sure using this hardcoded IP address '{0}' is safe here.";
-        private const int IPv4AddressParts  = 4;
+        private const int IPv4AddressParts = 4;
         private const string IPv4Broadcast = "255.255.255.255";
+        private const string OIDPrefix = "2.5.";
 
         private readonly string[] ignoredVariableNames =
-            {
-                "VERSION",
-                "ASSEMBLY",
-            };
-
-        private readonly DiagnosticDescriptor rule;
+        {
+            "VERSION",
+            "ASSEMBLY",
+        };
 
         protected abstract ILanguageFacade<TSyntaxKind> Language { get; }
-        protected abstract TSyntaxKind SyntaxKind { get; }
 
-        protected abstract string GetAssignedVariableName(TLiteralExpression stringLiteral);
+        protected abstract string GetAssignedVariableName(SyntaxNode stringLiteral);
         protected abstract string GetValueText(TLiteralExpression literalExpression);
-        protected abstract bool HasAttributes(TLiteralExpression literalExpression);
+        protected abstract bool HasAttributes(SyntaxNode literalExpression);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(rule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        protected DiagnosticDescriptor Rule { get; init; }
 
         protected HardcodedIpAddressBase(IAnalyzerConfiguration analyzerConfiguration) : base(analyzerConfiguration) =>
-            rule = Language.CreateDescriptor(DiagnosticId, MessageFormat);
+            Rule = Language.CreateDescriptor(DiagnosticId, MessageFormat);
 
         protected override void Initialize(SonarAnalysisContext context) =>
-            context.RegisterSyntaxNodeActionInNonGenerated(Language.GeneratedCodeRecognizer, CheckForHardcodedIpAddresses, SyntaxKind);
+            context.RegisterSyntaxNodeActionInNonGenerated(Language.GeneratedCodeRecognizer, CheckForHardcodedIpAddresses, Language.SyntaxKind.StringLiteralExpression);
+
+        protected bool IsHardcodedIp(string literalValue, SyntaxNode node) =>
+            literalValue != IPv4Broadcast
+            && IsNotAnObjectIdentifier(literalValue)
+            && IsRoutableNonLoopbackIPAddress(literalValue, out var address)
+            && (address.AddressFamily != AddressFamily.InterNetwork
+                || literalValue.Count(x => x == '.') == IPv4AddressParts - 1)
+            && (!(GetAssignedVariableName(node) is { } variableName)
+                || !ignoredVariableNames.Any(x => variableName.IndexOf(x, StringComparison.InvariantCultureIgnoreCase) >= 0))
```

Comment:
I understand it changed from early return in `CheckForHardcodedIpAddresses`  to a positive condition... But this is really difficult to read.

Initial logic:
```
(GetAssignedVariableName(stringLiteral) is { } variableName
 && ignoredVariableNames.Any(x => variableName.IndexOf(x, StringComparison.InvariantCultureIgnoreCase) >= 0))
```

IMO this is a regression in the overall readability of this logic.

the first part can be rewritten as `GetAssignedVariableName(node) is not { } variableName`. And either way the next part remains difficult to understand. Too many negations.

Could you please extract this into a local function with a good name and positive logic? Maybe `ShouldIgnoreVariableName`?